### PR TITLE
chore: add sync type and validation for orchestrator TaskSync

### PR DESCRIPTION
**Pull request by TBonnin**
Just the types and validation. No sync are yet scheduled or processed

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/orchestrator/lib/clients/validate.ts [hunk 1]
```diff
@@ -1,17 +1,64 @@
...
+    ...commonSchemaArgsFields
+});
+export const postConnectionArgsSchema = z.object({
+    type: z.literal('post-connection-script'),
+    postConnectionName: z.string().min(1),
+    fileLocation: z.string().min(1),
+    activityLogId: z.number().positive(),
+    input: jsonSchema,
+    ...commonSchemaArgsFields
+});
```
Comment (TBonnin) [inline packages/orchestrator/lib/clients/validate.ts:49]: adding syncArgsSchema and moving all the zod schema in `validate.ts` where they belong

File: packages/orchestrator/lib/clients/validate.ts [hunk 2]
```diff
@@ -1,17 +1,64 @@
...
+export const postConnectionArgsSchema = z.object({
+    type: z.literal('post-connection-script'),
+    postConnectionName: z.string().min(1),
+    fileLocation: z.string().min(1),
+    activityLogId: z.number().positive(),
+    input: jsonSchema,
```
Comment (bodinsamuel) [inline packages/orchestrator/lib/clients/validate.ts]: Can you remove input then? cc #2259

**Review by bodinsamuel — APPROVED** [review]
lgtm

File: packages/orchestrator/lib/clients/validate.ts (hunk 2 above)
Comment (TBonnin) [reply packages/orchestrator/lib/clients/validate.ts]: yep

**Pull request merged**